pt/lessons/specifics/mnesia.md

```diff
@@ -172,3 +179,119 @@ iex> data_to_read = fn ->
+
+```elixir
+iex> Mnesia.transaction(
+...>   fn ->
+...>     Mnesia.write({Person, 5, "Hans Moleman", "Ex-Mayor"})
+...>   end
+...> )
+```
+
+## Usando índices
+
+Mnesia suporta índices em colunas não chaves, e dados podem ser então consultados usando estes índices. Dessa forma podemos adicionar um índice na coluna `:job` da tabela `Person`:
+
+```elixir
+iex> Mnesia.add_table_index(Person, :job)
+{:atomic, :ok}
+```
+
+O resultado é similar ao retornado por `Mnesia.create_table/2`:
+
+- `{:atomic, :ok}` se a função executar com sucesso
+- `{:aborted, Reason}` se a função falhar
+
+Em particular, se o índice já existir, a razão será na forma `{:already_exists, table, attribute_index}`. Assim, se tentarmos adicionar este índice uma segunda vez teremos:
+
+```elixir
+iex> Mnesia.add_table_index(Person, :job)
+{:aborted, {:already_exists, Person, 4}}
+```
+
+Uma vez que o índice tenha sido criado com sucesso, nós podemos usá-lo para buscar uma lista de todos os diretores:
+
+```elixir
+iex> Mnesia.transaction(
+...>   fn ->
+...>     Mnesia.index_read(Person, "Principal", :job)
+...>   end
+...> )
+{:atomic, [{Person, 1, "Seymour Skinner", "Principal"}]}
+```
+
+## Combinação e seleção
+
+Mnesia suporta consultas complexas para recuperar dados de uma tabela na forma de combinação e funções de seleção ad-hoc.
+
+A função `Mnesia.match_object/1` retorna todos os registros que combinem com o padrão informado. Se qualquer coluna na tabela tiver índices, estes podem ser usados para tornar a busca mais eficiente. Use o átomo especial `:_` para identificar colunas que não devem participar da combinação.
+
+```elixir
+iex> Mnesia.transaction(
+...>   fn ->
+...>     Mnesia.match_object({Person, :_, "Marge Simpson", :_})
+...>   end
+...> )
+{:atomic, [{Person, 4, "Marge Simpson", "home maker"}]}
+```
+
+A função `Mnesia.select/2` permite que você especifique uma consulta customizada usando qualquer operador ou função da linguagem Elixir (ou Erlang, para esse efeito). Vamos olhar um exemplo que seleciona todos os registros que contém uma chave que é maior que 3:
+
+```elixir
+iex> Mnesia.transaction(
+...>   fn ->
+...>     {% raw %}Mnesia.select(Person, [{{Person, :"$1", :"$2", :"$3"}, [{:>, :"$1", 3}], [:"$$"]}]){% endraw %}
+...>   end
+...> )
+{:atomic, [[7, "Waylon Smithers", "Executive assistant"], [4, "Marge Simpson", "home maker"], [6, "Monty Burns", "Businessman"], [5, "Hans Moleman", "unknown"]]}
+```
+
+Vamos analisar isto. O primeiro atributo é a tabela, `Person`, e o segundo atributo é uma tripla da forma `{match, [guard], [result]}`:
+
+- `match` é o mesmo que você passaria para a função `Mnesia.match_object/1`. entretanto, note os átomos especiais `:"$n"` que especificam parâmetros posicionais que são usados no restante da consulta
+- a lista `guard` é uma lista de tuplas que especificam qual funções guarda aplicar, neste caso a função integrada `:>` (maior que) com o primeiro parâmetro posicional `:"$1"` e a constante `3` como atributos
```

Comment:
I checked the english version again and I think the following might be better?

EN: "the `guard` list is a list of tuples that specifies what guard functions to apply"
PT: "a lista `guard` é uma lista de tuplas que _especifica_ _quais_ funções guarda aplicar"